Enhance domain consistency in My Home

vykes-mac opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to https://github.com/Automattic/wp-calypso/issues/99237

The purpose of this PR is to make the domain displayed throughout the onboarding consistent. Whenever the user purchase a custom domain that domain is shown in the fullscreen launchpad but the `.wordpress.com` subdomain is shown on the My Home Dashboard. This happens because the domain take sometime to become active in the backend, particularly when activating ssl on the domain, after the domain completes activation/setup then all the sites urls are updated with the domain, before that the sites url will be the `.wordpress.com` subdomain. This PR favors the custom domain over the siteSlug since the domain is available but just not finish setting up. **Note** when the user clicks the `View site` CTA on the Launch celebration notice we still have to maintain `.wordpress.com` subdomain until the domain ssl finishes otherwise the site will not be found when the CTA us clicked. 

## Proposed Changes
* Added loading state for site domain display in the `Site` component.
* Updated domain rendering logic to prioritize custom domains over default site domain.
* Refactored CelebrateLaunchModal to use the correct site domain based on custom domains.

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

* As part of the overall onboarding experience improvement.

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* go to `/setup/onboarding` 
* Choose a paid domain, personal plan and complete the flow quickly 
* Verify the that your custom domain is shown in My Home sidebar instead of `.wordpress.com` subdomain
* Launch your site immediately and verify the custom domain is shown in the celebration modal instead of wordpress subdomain.
* Clicking `View site` should take you to your site but the url will be `.wordpress.com` subdomain until the domain setup is completed in the backend.


**Before**
p1738596412795819/1738593870.964859-slack-C06FCT6EEHH

**After**

https://github.com/user-attachments/assets/12bc939a-c8a6-4c43-8d93-926a48396b92


## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

vykes-mac committed:
Enhance site domain on My Home

* Added loading state for site domain display in the Site component.
* Updated domain rendering logic to prioritize custom domains over default site domain.
* Introduced new styles for loading state in site domain.
* Refactored CelebrateLaunchModal to use the correct site domain based on custom domains.

paulopmt1 commented on client/my-sites/customer-home/components/celebrate-launch-modal.jsx:
```diff
@@ -21,11 +21,10 @@ function CelebrateLaunchModal( { setModalIsOpen, site, allDomains } ) {
-	const hasCustomDomain = Boolean(
-		transformedDomains.find( ( domain ) => ! domain.isWPCOMDomain )
-	);
+	const customDomains = transformedDomains.filter( ( domain ) => ! domain.isWPCOMDomain );
```
Nice! Can we add a unit test for these changes?

paulopmt1 commented on client/blocks/site/index.jsx:
```diff
@@ -112,32 +113,44 @@ class Site extends Component {
+		const siteDomain = customDomains.length > 0 ? customDomains[ 0 ].domain : site.domain;
 
 		return (
 			<div className="site__domain">
 				{ isJetpackCloud() &&
 					homeLink &&
 					translate( 'View %(domain)s', {
-						args: { domain: site.domain },
+						args: { domain: siteDomain },
 					} ) }
-				{ ( ! isJetpackCloud() || ! homeLink ) && site.domain }
+				{ ( ! isJetpackCloud() || ! homeLink ) && siteDomain }
 			</div>
 		);
 	};
 
 	renderDomainAndInlineBadges = () => {
-		const { site, homeLink, translate } = this.props;
+		const { site, homeLink, translate, customDomains, isLoadingDomains } = this.props;
+
+		if ( isLoadingDomains ) {
+			return <div className="site__domain is-loading" />;
+		}
+
+		const siteDomain = customDomains.length > 0 ? customDomains[ 0 ].domain : site.domain;
```
It's working as expected! A unit test for them would also be great

paulopmt1 reviewed (approved):
Good job @vykes-mac, everything is working as expected!

- [x] Verify the that your custom domain is shown in My Home sidebar instead of .wordpress.com subdomain
- [x] Launch your site immediately and verify the custom domain is shown in the celebration modal instead of wordpress subdomain.
- [x] Clicking View site should take you to your site but the url will be .wordpress.com subdomain until the domain setup is completed in the backend.


| Before | After |
|--------|--------|
| ![image](https://github.com/user-attachments/assets/d7fd32f8-c05e-4e13-9030-118066ecbc4c) | ![image](https://github.com/user-attachments/assets/80c631be-32f7-488c-8636-a470d60f1c18) | 
| ![image](https://github.com/user-attachments/assets/ade6ede4-af1d-4e98-8d66-427db4f27e3b) | ![image](https://github.com/user-attachments/assets/b435a7ec-3926-496f-b56f-d2ef07429a39) |
| ![image](https://github.com/user-attachments/assets/eeadda03-5795-4b99-b0a6-d306d1b019ab) | ![image](https://github.com/user-attachments/assets/9e942ec3-7906-4a0f-8ece-c32d01ea2543) | 

Left some non-blocking comments in the PR.

vykes-mac committed:
Add unit tests for Site component domain rendering

vykes-mac committed:
Add unit tests for CelebrateLaunchModal component

vykes-mac committed:
Merge branch 'trunk' into update/domain-consistency-on-home

vykes-mac committed:
Enhance domain loading logic in site index component

- Import isRequestingSiteDomains selector to improve domain loading state management.
- Update isLoadingDomains calculation to consider both loaded state and request status.

vykes-mac committed:
Fix tests

p-jackson reviewed (approved):
Works great. I tested with purchasing zero and two domains during onboarding just to double check. I like the rationale of using the custom domain when it's just for display, and the site's official domain when it actually needs to be visited.

I'm approving, but one other place which could be fixed is this panel in My Home

![CleanShot 2025-03-21 at 19 58 39@2x](https://github.com/user-attachments/assets/27589eea-c823-44be-b7c4-09d97cc7dd32)

vykes-mac commented:
> Works great. I tested with purchasing zero and two domains during onboarding just to double check. I like the rationale of using the custom domain when it's just for display, and the site's official domain when it actually needs to be visited.
> 
> I'm approving, but one other place which could be fixed is this panel in My Home

vykes-mac committed:
Enhance site preview component to display custom domains

- Introduced logic to fetch and display custom domains for the selected site.
- Updated domain rendering to prioritize non-WPCOM domains.
- Adjusted the site preview to reflect the selected site's domain accurately.
